src/main/java/org/isf/permissions/service/PermissionIoOperations.java

```diff
@@ -36,79 +33,33 @@
 	}
 
-	public List<Permission> retrievePermissionsByCurrentLoggedInUser(String currentUserName) throws OHServiceException {
+	public List<Permission> retrievePermissionsByCurrentLoggedInUser(String currentUserName) {
 		return repository.retrievePermissionsByCurrentLoggedInUser(currentUserName);
 	}
 
-	public Permission retrievePermissionById(Integer id) throws OHServiceException {
+	public Permission retrievePermissionById(Integer id) {
 		return repository.findById(id).orElse(null);
 	}
 
-	public Permission retrievePermissionByName(String name) throws OHServiceException {
+	public Permission retrievePermissionByName(String name) {
 		return repository.findByName(name);
 	}
 
-	public Permission insertPermission(Permission permission) throws OHServiceException {
-		Permission permissionResult = repository.save(permission);
-		permission.getGroupPermission().forEach(gp -> {
-			gp.setPermission(permissionResult);
-			groupPermissionRepository.save(gp);
-		});
-		return permissionResult;
-	}
-
-	public Permission updatePermission(Permission permission) throws OHServiceException {
-		// All group permissions (could exist already in the DB)
-		List<GroupPermission> gp = permission.getGroupPermission();
-
-		Permission permissionUpdated = repository.save(permission);
-		// retrieve groupPermission stored in DB
-		List<GroupPermission> groupPermissionInDB = groupPermissionRepository.findByPermission_id(permission.getId());
-
-		// calculate GroupPermission to delete
-		List<String> allUserGroupCodesToStore = gp.stream().map(item -> item.getUserGroup().getCode()).collect(Collectors.toList());
-		List<String> allUserGroupCodesOnDB = groupPermissionInDB.stream().map(item -> item.getUserGroup().getCode()).collect(Collectors.toList());
-
-		List<String> allUserGroupCodesToDelete = allUserGroupCodesOnDB.stream().filter(onDB -> !allUserGroupCodesToStore.contains(onDB)).collect(Collectors.toList());
-		List<String> allUserGroupCodesToInsert = allUserGroupCodesToStore.stream().filter(item -> !allUserGroupCodesOnDB.contains(item)).collect(Collectors.toList());
-
-		// delete obsolete relations
-		List<GroupPermission> groupPermissionToDelete = groupPermissionRepository.findByUserGroup_codeInAndPermission_id(allUserGroupCodesToDelete, permission.getId());
-		groupPermissionRepository.deleteAll(groupPermissionToDelete);
-
-		// store new relations
-		gp.forEach(item -> {
-			if (allUserGroupCodesToInsert.contains(item.getUserGroup().getCode())) {
-				item.setPermission(permissionUpdated);
-				groupPermissionRepository.save(item);
-			}
-		});
-		return repository.getReferenceById(permissionUpdated.getId());
-	}
-
-	public void deletePermission(Integer id) throws OHServiceException {
-		repository.deleteById(id);
-	}
-
-	public List<Permission> retrieveAllPermissions() throws OHServiceException {
+	public List<Permission> retrieveAllPermissions() {
 		return repository.findAll();
 	}
 
 	public boolean exists(int id) throws OHServiceException {
```

Comment:
Why is the `throws` clause here and remove on all other methods?   I don't think it can be thrown here.